Sources/AuthFoundation/AuthFoundation.docc/ManagingUserCredentials.md

```diff
@@ -49,7 +49,45 @@ NotificationCenter.default.addObserver(forName: .defaultCredentialChanged,
 
 ## Managing multiple credentials
 
-Changing the ``Credential/default`` property doesn't remove the old value from storage. Instead, all credentials that are stored are available through the ``Credential/allCredentials`` property. This can enable multiple user credentials to be used simultaneously.
+Changing the ``Credential/default`` property doesn't remove the old value from storage. Instead, all credentials that are stored are available. This can enable multiple user credentials to be used simultaneously.
+
+### Working with multiple credentials
+
+Several approachs to finding and using credentials are provided, depending upon your application use-case. These facilities simplifies the process of assigning different credentials to different tasks within your application.
+
+#### Finding credentials by ID
+
+All tokens are automatically assigned a unique ID it can be identified by. This can be seen in the ``Token/id`` (as well as through the ``Credential/id`` property). This identifier may be used with the ``Credential/with(id:)`` function.
+
+```swift
+if let credential = try Credential.with(id: serviceTokenId) {
+    // Do something with the credential
+}
+```
+
+#### Finding credentials by developer-assigned tags
+
+When storing tokens, the ``Credential/store(token:tags:)`` function accepts an optional collection of tags you can use to identify the purpose of different tokens.  The ``Credential/find(where:)`` function allows you to query tokens based on those tags at a later date.  Furthermore, those tags can later be updated by changing the credential's ``Credential/tags`` property.
+
+```swift
+try Credential.store(token: newToken, tags: ["service": "purchase"])
+
+// Later ...
+
+if let credential = try Credential.find(where: { $0.tags["service"] == "purchase" }).first {
+    // Use the credential
+}
+```
+
+#### Finding credentials by ID Token claims
+
+If a token contains a valid ``Token/idToken``, the claims it represents are available within the ``Credential/find(where:)`` expression. The object returned within that expression supports the ``HasClaims`` protocol, meaning its values can be queried within that filter.
+
+```swift
+let userCredentials = try Credential.find { metadata in
```

Comment:
does this throw if not found? I'm missing a bit of context around `Credential` and why we are using class functions here instead of instance.

Reply:
It throws if there is an error thrown within the underlying storage mechanism.  For example, deserialization fails, Keychain operations fail, etc.

Reply:
To answer your other question, these are class functions since at the point a developer would be using this, they wouldn't have an instance of anything to start from.  AuthenticationFlows are only created for the purpose of authenticating, but throughout the lifetime of an app, that only accounts for a brief moment. The majority of the time, the developer will be interacting with credentials.

Plus, they may have multiple credentials for different purposes (some only used infrequently).  So these methods are used to query for / find the credential they need within that part of their application.